sql/planbuilder/scalar.go

```diff
@@ -212,16 +212,28 @@ func (b *Builder) buildScalar(inScope *scope, e ast.Expr) sql.Expression {
 		}
 		return expression.NewCollatedExpression(innerExpr, collation)
 	case *ast.ValuesFuncExpr:
-		col := b.buildScalar(inScope, v.Name)
-		fn, err := b.cat.Function(b.ctx, "values")
-		if err != nil {
-			b.handleErr(err)
-		}
-		values, err := fn.NewInstance([]sql.Expression{col})
-		if err != nil {
-			b.handleErr(err)
+		if b.insertActive {
```

Comment:
I'm seeing an increasing reliance on this pattern in the new analyzer, it worries me a bit

If you're going to go down this route, there should be beefier unit tests of the plan builder methods, otherwise this becomes pretty hard to reason about